core/src/test/scala/org/http4s/client/jdkhttpclient/JdkWSClientSpec.scala

```diff
@@ -33,162 +31,160 @@ import org.http4s.websocket.WebSocketFrame
 import org.java_websocket.WebSocket
 import org.java_websocket.handshake.ClientHandshake
 import org.java_websocket.server.WebSocketServer
-import org.specs2.mutable.Specification
 import scodec.bits.ByteVector
 
-import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent.duration._
+import munit.CatsEffectSuite
 
-class JdkWSClientSpec extends Specification with CatsEffect {
-  implicit val timer: cats.effect.Timer[IO] = IO.timer(global)
-  implicit val cs: cats.effect.ContextShift[IO] = IO.contextShift(global)
+class JdkWSClientSpec extends CatsEffectSuite {
 
-  val webSocket: WSClient[IO] = JdkWSClient.simple[IO].unsafeRunSync()
+  def webSocket = ResourceFixture(JdkWSClient.simple[IO])
```

Comment:
If I change `def` to `val` here, I get exceptions for all but the first test like this:

```
java.lang.IllegalStateException: Promise already completed.
    at scala.concurrent.Promise.complete(Promise.scala:53)
    at scala.concurrent.Promise.complete$(Promise.scala:52)
    at scala.concurrent.impl.Promise$DefaultPromise.complete(Promise.scala:187)
    at scala.concurrent.Promise.success(Promise.scala:86)
    at scala.concurrent.Promise.success$(Promise.scala:86)
    at scala.concurrent.impl.Promise$DefaultPromise.success(Promise.scala:187)
    at munit.CatsEffectFunFixtures$ResourceFixture$.$anonfun$apply$4(CatsEffectFunFixtures.scala:50)
    at cats.effect.IOFiber.mapK(IOFiber.scala:970)
    at cats.effect.IOFiber.succeeded(IOFiber.scala:794)
    at cats.effect.IOFiber.runLoop(IOFiber.scala:257)
    at cats.effect.IOFiber.execR(IOFiber.scala:908)
    at cats.effect.IOFiber.run(IOFiber.scala:135)
    at cats.effect.unsafe.WorkerThread.runFiber(WorkerThread.scala:162)
    at cats.effect.unsafe.WorkerThread.run(WorkerThread.scala:295)
```

Is this expected?

Reply:
That surprises me, too. That's a good upstream question.

Reply:
This has been fixed 0.13.0